### ISSUE TITLE
web-server dockerfile: add compression

### ISSUE THREAD

**Issue by syphar** [rust-lang/docs.rs#2271]
In my mind, this could be solved via: 
- adding a reverse proxy to the container (NGINX, Caddy) and enabling compression 
- adding a compression middleware to axum
 
I assume axum is fine being exposed to the internet and doesn't need to be behind a reverse proxy (?). 

Also in our current NGINX config: IP blocks, while it's unclear if we would handle this in AWS / CloudFront, or if we would have to handle this in our webserver (see https://github.com/rust-lang/simpleinfra/issues/353)

**Comment by Nemo157** [comment]
We'd also need to have TLS in the container if it is exposed to the internet, but I assume we're still having it only accessed by cloudfront over a private network so that shouldn't be necessary.

**Comment by syphar** [comment]
Yep, that's my thinking, our webservers would only be accessed through CloudFront, so we don't have to do TLS

**Comment by syphar** [comment]
And especially between the US & CF POPs in europe / asia compression definitely makes a difference

**Comment by tverghis** [comment]
Hi! I'd like to take a stab at solving this by adding a compression layer to the middleware stack in the web server. If that's ok, I have a couple of questions:

1. Can this compression middleware be applied to all routes served by the web server, or is there a reason to target specific routes?
2. I'm planning on using [`tower_http::compression`](https://docs.rs/tower-http/latest/tower_http/compression/index.html), which supports `brotli`, `deflate`, `gzip` or `zstd`. I don't know enough about this field to make a really educated decision on which one to choose. Does anyone have advice? I know that `gzip` is widely supported/used, and performs reasonably well. Should we start there?

**Comment by syphar** [comment]
When using tower-http compression, see the comments / change here : https://github.com/rust-lang/crates.io/pull/7330/files

**Comment by syphar** [comment]
Having read a little more about it I would say we should enable brotli, when easily possible. 

We first would add this to our webserver, and then probably need to change CloudFront config to support this, or use it between the browser and the CDN POP.
